src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDAO.java

```diff
@@ -11,8 +11,12 @@
 
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public interface ShadowManagerDAO {
+    ExecutorService dbWriteThreadPool = Executors.newCachedThreadPool();
```

Comment:
call shutdown, but not shutdownNow when closing shadow manager. This will prevent new threads from spawning.